tests/testthat/test-as-data-frame.R

```diff
@@ -122,6 +122,140 @@ with_mock_crunch({
         expect_identical(.crunchPageSize(ds$starttime), 100000L)
         expect_identical(.crunchPageSize(2016 - ds$birthyr), 50000L)
     })
+    
+    test_that("can manipulate the order of a crunchDataFrame", {
+        ds_df <- as.data.frame(ds)
+        gndr <- ds_df$v1
+        new_order <- c(4,3,2,1)
+        ds_df$.order <- new_order
+        expect_equal(ds_df$v1, gndr[new_order])
+        expect_equal(nrow(ds_df), length(new_order))
+        ds_df2 <- as.data.frame(ds, selection = new_order)
+        expect_equal(ds_df2$v1, gndr[new_order])
+        expect_equal(nrow(ds_df2), length(new_order))
+    })
+    
+    test_that("merge.CrunchDataFrame works", {
+        ds_df <- as.data.frame(ds)
+        local_df <- data.frame(gender=c("Male", "Female"), new="new")
+        expect_silent(merged_df <- merge(ds_df,
+                                         local_df,
+                                         by.x = "gender",
+                                         by.y = "gender"))
+        expect_identical(nrow(merged_df), nrow(ds_df))
+        expect_identical(merged_df$new,
+                         factor(c("new", "new", NA, "new", "new", "new", "new",
+                                  NA, NA, "new", "new", "new", "new", NA, NA,
+                                  NA, "new", "new", "new", NA, "new", "new",
+                                  "new", NA, "new")))
+        expect_error(merged_df <- merge(ds_df,
```

Comment:
I personally like to factor out the input validation tests to their own `test_that`. And keep in mind that the more descriptive the label in the `test_that`, the easier it is to read test output and understand what failed without having to look up line numbers etc.